File: Src/DragqnLD/DragqnLD.Core/Implementations/Query/QueryDefinition.cs
using DragqnLD.Core.Interfaces.Query;
using VDS.RDF.Query;

namespace DragqnLD.Core.Implementations
{
    public class QueryDefinition : IQueryDefinition
    {
        public QueryDefinition(string name, string description, SparqlParameterizedString constructQuery, string constructQueryUriParameterName, SparqlQuery selectQuery)
        {
            Name = name;
            Description = description;
            ConstructQuery = constructQuery;
            ConstructQueryUriParameterName = constructQueryUriParameterName;
            SelectQuery = selectQuery;
        }

        public string Name { get; set; }
        public string Description { get; set; }
        public SparqlParameterizedString ConstructQuery { get; set; }
        public string ConstructQueryUriParameterName { get; set; }
        public SparqlQuery SelectQuery { get; set; }
    }
}

File: Src/DragqnLD/DragqnLD.Web.Html5/App/models/queryDefinition.ts
import sparqlQueryInfo = require("models/sparqlQueryInfo");

class queryDefinition {
    id = ko.observable<string>().extend({ required: true });
    name = ko.observable<string>().extend({ required: true });
    description = ko.observable<string>().extend({ required: true });
    constructQueryUriParameterName = ko.observable<string>().extend({ required: true });
    constructQuery: sparqlQueryInfo;
    selectQuery: sparqlQueryInfo;

    constructor(dto: queryDefinitionDto) {

        this.id(dto.Id);
        this.name(dto.Name);
        this.description(dto.Description);
        this.constructQueryUriParameterName(dto.ConstructQueryUriParameterName);
        this.constructQuery = new sparqlQueryInfo(dto.ConstructQuery);
        this.selectQuery = new sparqlQueryInfo(dto.SelectQuery);
    }

    public static empty(): queryDefinition {
        return new queryDefinition({
            Id: "",
            Name: "",
            Description: "",
            ConstructQueryUriParameterName: "",
            ConstructQuery: {
                Query: "",
                DefaultDataSet: "",
                SparqlEndpoint: ""
            },
            SelectQuery: {
                Query: "",
                DefaultDataSet: "",
                SparqlEndpoint: ""
            }
        });
    }

    public toDto(): queryDefinitionDto {
        return {
            Id: this.id(),
            Name: this.name(),
            Description: this.description(),
            ConstructQueryUriParameterName: this.constructQueryUriParameterName(),
            ConstructQuery: this.constructQuery.toDto(),
            SelectQuery: this.selectQuery.toDto()
        };
    }
}

export = queryDefinition;